javav2/example_code/ecr/src/main/java/com/example/ecr/scenario/ECRActions.java

```diff
@@ -0,0 +1,555 @@
+        String policyDocumentTemplate = """
+             {
+              "Version" : "2012-10-17",
+              "Statement" : [ {
+                "Sid" : "new statement",
+                "Effect" : "Allow",
+                "Principal" : {
+                  "AWS" : "%s"
+                },
+                "Action" : "ecr:BatchGetImage"
+              } ]
+            }
+             """;
+
+        String policyDocument = String.format(policyDocumentTemplate, iamRole);
+        SetRepositoryPolicyRequest setRepositoryPolicyRequest = SetRepositoryPolicyRequest.builder()
+            .repositoryName(repoName)
+            .policyText(policyDocument)
+            .build();
+
+        CompletableFuture<SetRepositoryPolicyResponse> response = getAsyncClient().setRepositoryPolicy(setRepositoryPolicyRequest);
+        response.whenComplete((resp, ex) -> {
+            if (resp != null) {
+                System.out.println("Repository policy set successfully.");
+            } else {
+                Throwable cause = ex.getCause();
+                if (cause instanceof InvalidParameterException) {
+                    InvalidParameterException e = (InvalidParameterException) cause;
+                    System.out.format("Error setting repository policy for repository: %s. The IAM role '%s' is invalid. %s", repoName, iamRole, e.getMessage());
+                } else if (cause instanceof RepositoryPolicyNotFoundException) {
+                    RepositoryPolicyNotFoundException e = (RepositoryPolicyNotFoundException) cause;
+                    System.out.format("Error setting repository policy for repository: %s. The repository policy does not exist. %s", repoName, e.getMessage());
+                } else if (cause instanceof EcrException) {
+                    EcrException e = (EcrException) cause;
+                    System.out.format("Unexpected error setting repository policy for repository: %s. %s", repoName, e.getMessage());
+                } else {
+                    System.err.println("Unexpected error occurred: " + ex.getMessage());
+                }
+            }
+        });
+
+        // Wait for the CompletableFuture to complete
+        response.join();
+    }
+    // snippet-end:[ecr.java2.set.repo.policy.main]
+
+    // snippet-start:[ecr.java2.create.repo.main]
+    /**
+     * Creates an Amazon Elastic Container Registry (Amazon ECR) repository.
+     *
+     * @param repoName the name of the repository to create.
+     * @return the Amazon Resource Name (ARN) of the created repository, or an empty string if the operation failed.
+     * @throws IllegalArgumentException if the repository name is null or empty.
+     * @throws EcrException             if an error occurs while creating the repository.
+     */
+    public String createECRRepository(String repoName) {
+        if (repoName == null || repoName.isEmpty()) {
+            throw new IllegalArgumentException("Repository name cannot be null or empty");
```

Comment:
This is a case where I think "System.exit" can be appropriate. The user provided a bad name and it's obvious where a failure is, so a stack trace isn't necessary.

Reply:
The calling code will capture htis and end program.